client/CPlayerInterface.cpp

```diff
@@ -230,7 +232,7 @@ void CPlayerInterface::performAutosave()
 					+ std::to_string(cb->getDate(Date::WEEK))
 					+ std::to_string(cb->getDate(Date::DAY_OF_WEEK));
 
-			cb->save("Saves/" + prefix + "Autosave_" + stringifiedDate);
+			cb->save("Saves/Autosave/" + prefix + "Autosave_" + id + "_" + stringifiedDate);

```

Comment:
If prefix has folder name which is map+date, do we need id here?
I mean, here we have:
```
Saves/Autosave/mymap_datetime/Autosave_abcd_111
Saves/Autosave/mymap_datetime/Autosave_abcd_112
```
Can we have just `Saves/Autosave/mymap_datetime/Autosave_111` ?
And if so, do we need generated and serialised uuid at all?

Reply:
Think that it is just configurable. You can configure it so that it ends up in a folder. Then it is necessary.

Reply:
According to lines 215-217 prefix cannot be empty. I barely imagine how users can use savePrefix without changing its logic. If I put `abc` into savePrefix, we won't have separate folders.
Then it turns into strange tool to disable folders for each game, which could be useful (especially on mobile devices as each save takes 2mb, so if I play 100 times, I will have 20*5 saves which is 200mb) but probably better to have dedicated parameter and remove savePrefix parameter.
But maybe I don't understand something about logic of this parameter

Reply:
Check the variable `usePrefix` and maybe check the gui. :)

If you uncheck the checkbox every save file is in one folder (`Autosave`). No problem, if limit > 0 (overwriting active). But if limit = 0 (keeping all) the filename conflicts without id.

Reply:
In current implementation we have overrides with both enabled/disabled prefix with both limit 0 and >0.
I played several times on same map with different configuration and here what I have.

<img width="229" alt="Снимок экрана 2023-08-23 в 04 32 09" src="https://github.com/vcmi/vcmi/assets/9308612/9dbd6e5f-3382-47a7-b566-5778acee34c7">

Now in your implementation there will be following:
```
usePrefix=true, savePrefix="", autosaveCountLimit=0

Autosaves/mymap_23082023_103120/Autosave_abcd_111
Autosaves/mymap_23082023_103120/Autosave_abcd_112
```

Then I start new map
```
Autosaves/mymap_23082023_121222/Autosave_abcd_111
Autosaves/mymap_23082023_121222/Autosave_abcd_111
```

No overrides because I'd very hard to start map again in exacltly same moment.

The only scenario I see with possible overrides is
```
usePrefix=true, savePrefix="myprefix", autosaveCountLimit=0
```
But this is current behavior and not a regression. That's why intention to have extra uuid in filename is not clear

Reply:
Case 1 (overwrite possible):
`usePrefix=true, savePrefix="myprefix", autosaveCountLimit=0`
`usePrefix=false, savePrefix="", autosaveCountLimit=0`

Case 2 (no overwrite possible):
`usePrefix=true, savePrefix="", autosaveCountLimit=0`

Case 3 (overwrite intended):
`autosaveCountLimit>0`

With the current solution case 1 & 2 get the additional id to prevent overwriting. Also (as written) the uuid is a preparation for cleaning up.

Yes this is the current behavior. We can change it completely. But I don't want to invest time for that now.

Reply:
> Case 1 (overwrite possible):

id is not needed

> Case 2 (no overwrite possible):

Covered by date-time, id is also not needed

> Case 3

id is not needed